src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,42 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  margin: 0 auto;
+  width: calendarSize($cell-size, $padding, $gap);
+  gap: $gap;
+  padding: $padding;
+
+  @each $day, $index in $days {
+    &--start-day--#{$day} :first-child {
+      margin-left: ($cell-size + $gap) * $index;
+    }
+  }
+
+  @for $monthLength from 28 through 31 {
+    &--month-length--#{$monthLength} :nth-child(n + #{$monthLength + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    @include square;
+    border: 1px solid black;
+    line-height: $cell-size;
+    text-align: center;
+    background-color: #eee;
+
+    transition: transform 0.3s, background-color 0.3s;
```

Comment:
Wrong duration - needs to be half of a second.
You can create a variable for it.